replaces/README.md

```diff
@@ -0,0 +1,154 @@
+
+1. All the data sources involved must share the partition key.
+
+1. The update process is atomic and we are able to properly handle incoming changes during the process if you append data using the URL parameter.
+
+
+In the following sections, you'll find several examples of how to do replacements to maintain all your data consistent.
+
+## 1-level dependencies
+
+The most common use case for a replacement is:
+
+![basic_uc](img/basic_uc.png)
+
+In this case, you can create the resources and load data: 
+
+```sh
+tb push pipes/mv_ds_a_b.pipe --push-deps
+tb datasource append ds_a datasources/fixtures/a.csv
+```
+
+This will load the data in ds_a and it will be automatically populated to `ds_b`.
+
+Now, we want to replace some data. Let's do it and check how it is automatically done in the dependent-view `ds_b`.
+
+Our data source `ds_a` has two columns where both columns have the same data except `a1=1`. This row is wrong, we can update this record and it will be automatically updated in `ds_b`.
+
+```sh
+tb datasource replace ds_a --sql-condition="a1=1" datasources/fixtures/a_replacement.csv
+```
+
+You can now check how it has been fixed in both data sources.
+
+## N-level dependencies
+
+As we mentioned before, when you do a replacement just the first level dependency will be automatically updated.
+
+![n-level_dependencies](img/n-level-dependencies.png)
+
+If we push the new resources and load the data we can check how data is properly populated in all the data sources.
+
+```sh
+tb push pipes/mv_ds_b_c.pipe --push-deps
+tb push pipes/mv_ds_c_d.pipe --push-deps
+tb datasource truncate ds_a
+tb datasource truncate ds_b
+tb datasource append ds_a datasources/fixtures/a.csv
+```
+
+However, we have the same problem we had before. One of the rows `a1=1` is wrong and we need to fix it.
+
+```sh
+tb datasource replace ds_a --sql-condition="a1=1" datasources/fixtures/a_replacement.csv
+```
+
+If you check your data sources you can verify just `ds_a` and `ds_b` contains the new data. What can we do with the other data sources??
+
+You can use different strategies. Our recommendation is to use the materialization pipes to export data and do a replacement in the next node. In our example, we have four data sources `ds_a`, `ds_b`, `ds_c`, and `ds_d`. The first two ones are already fixed. Then we need to fix C and D: 
+1. Clone the materialization pipe `mv_ds_b_c`.
+1. Expose the latest node as an endpoint. It will export data already fixed.
+1. Load the data into `ds_c` doing a replacement. It will update the data into `ds_c`and `ds_d`.
+
+```sh
+tb push endpoints/mv_ds_b_c_export.pipe
+
+curl -H "Authorization: Bearer $TOKEN" -d "select * from mv_ds_b_c_export where a1=1 format CSVWithNames" "https://api.tinybird.co/v0/sql" > datasources/fixtures/updates.csv
```

Comment:
We should add a note that this only would work if the exported data is < 100 MB as we currently have this limit when exporting.

Reply:
Btw you could also do something like

```
curl \
  -H "Authorization: Bearer <DATASOURCES:CREATE your_token>" 
  -X POST https://api.tinybird.co/v0/datasources \
  -d url='https://api.tinybird.co/v0/pipes/batcher_pipe.csv?token=p.eyJ1IjogIjZmMDU5NzBjLWY3MGMtNDkyMy04Y2ViLWE5NWYxNTlhOTk1NSIsICJpZCI6ICJiMTQwNDIzNS0zNDYxLTQ5ZjctODMxMS1jZWMwMmU0NzNmOTUifQ.is-u1CMFjRUeUvYhI_bFPIKsl1ZayIFLncD09G_OWUw' \
  -d mode=replace \
  -d name=events \
  -d replace_condition="id IN (3,4,5,6,7,)'"
  ```
  
  Where the URL is a Tinybird endpoint